### PR TITLE
:sparkles: Add pprof option for global hub components

### PR DESCRIPTION
**Pull request by yanmxa**
## Summary

Add the pprof tool options for footprint investigate

## Related issue(s)

Fixes # https://github.com/stolostron/multicluster-global-hub/pull/849

### CODE REVIEW

**Comment by yanmxa** [comment]
override ci/prow/sonarcloud
override "SonarCloud Code Analysis"

File: manager/cmd/manager/main.go [hunk 1]
```diff
@@ -299,6 +301,15 @@ func doMain(ctx context.Context, restConfig *rest.Config) int {
 		setupLog.Error(err, "failed to complete configuration")
 		return 1
 	}
+
+	if managerConfig.EnablePprof {
+		go func() {
+			if err := http.ListenAndServe("localhost:6060", nil); err != nil {
```
Comment (clyang82) [inline manager/cmd/manager/main.go:307]: Do not we need to register the endpoint and handler for the pprof?

Comment (clyang82) [reply manager/cmd/manager/main.go:307]: BTW, we may expose `:6060` insteadof `localhost:6060`.

Comment (yanmxa) [reply manager/cmd/manager/main.go:307]: > Do not we need to register the endpoint and handler for the pprof?

we don't need to register the handler by default.
`_ "net/http/pprof"`: the package is typically only imported for the side effect of registering its HTTP handlers. The handled paths all begin with /debug/pprof/. reference: https://pkg.go.dev/runtime/pprof#Profile

Comment (yanmxa) [reply manager/cmd/manager/main.go]: > BTW, we may expose :6060 insteadof localhost:6060.

Done.

**Comment by clyang82** [comment]
please take a look at [E Security Rating on New Code](https://sonarcloud.io/dashboard?id=open-cluster-management_hub-of-hubs&pullRequest=907)

**Review by clyang82 — APPROVED** [review]
LGTM

**Pull request merged**